.build/build-sonar.xml

```diff
@@ -0,0 +1,167 @@
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
+<!--
+  ~ Licensed to the Apache Software Foundation (ASF) under one
+  ~ or more contributor license agreements.  See the NOTICE file
+  ~ distributed with this work for additional information
+  ~ regarding copyright ownership.  The ASF licenses this file
+  ~ to you under the Apache License, Version 2.0 (the
+  ~ "License"); you may not use this file except in compliance
+  ~ with the License.  You may obtain a copy of the License at
+  ~
+  ~     http://www.apache.org/licenses/LICENSE-2.0
+  ~
+  ~ Unless required by applicable law or agreed to in writing, software
+  ~ distributed under the License is distributed on an "AS IS" BASIS,
+  ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  ~ See the License for the specific language governing permissions and
+  ~ limitations under the License.
+  -->
+<project basedir="." name="apache-cassandra-sonar-tasks" xmlns:if="ant:if" xmlns:unless="ant:unless">
+    <property name="sonar-scanner-cli.version" value="5.0.1.3006-linux"/>
+    <property name="sonar.workdir" value="${build.dir}/sonar"/>
+    <property name="sonar.download.dir" value="${user.home}/.cassandra-cache/tools"/>
+    <property name="sonar-scanner.home" value="${sonar.download.dir}/sonar-scanner-${sonar-scanner-cli.version}"/>
```

Comment:
I think we should put this directory somewhere under the project root, right? I would not expect the project to change anything in my home directory.

Reply:
This is a cache. Putting that under the build directory in the project invalidates that. That is, it would no longer be a cache, and everything would have to be fetched again whenever one does `ant clean`. We will have a similar story with OWASP analysis which needs to fetch much more data to run. I'd even consider putting the Sonar volumes there because it is probably not the best thing to have them wiped on `ant clean`, especially when the server is still running. Though, it is something that deserves more discussion...

Reply:
@jacek-lewandowski maybe putting it all to `/tmp` would be OK too? That would still cache it, kind of, even though it would not survive the restart of the developer's PC.

Reply:
I am -1 on polluting the home directory and you can't rely on literal '/tmp'  on all machines, but tmp.dir may work.

Reply:
I see your point and I don't want to make anything that would break working with multiple project worktrees, hence we are discussing.

Though:
1. the server needs to be running in order to browse the results, there can be an Ant task to stop it
2. When working with multiple worktrees, starting a new server for each worktree could be confusing, we can easily confuse which port was used for which project worktree
3. We do not necessarily setup volumes assuming that we will not reuse the results later, this would fix the problem of storing data in the build directory and exposing to problems when trying to clean the build directory while the server is still working. We would not need to store the data outside the project worktree. 
4. In order to reuse the server for multiple project worktrees, we could create a project in Sonar per project worktree. The only question would be the name of the project - branch name would not be sufficient because, as you mentioned, you operate on more than one `trunk`, which is totally justified. Perhaps branch name and sha? Or Git worktree name? 
5. OWASP cache will be more like Maven repository, I don't think we would want to clean it because it is branch-agnostic. It also ~ 500MB database.

wdyt?

Reply:
Some ideas @smiklosovic for 4 then?

Reply:
branch name and SHA seems like a good combination, SHA might be trimmed to e.g. first 6 characters for brevity?

Reply:
Agree with @driftx , we can't be polluting home. (And it's not even in $HOME/.cache anyway)

This should follow the existing precedences, see build-owasp.xml and build-resolver.xml .  Use the build dir or the correct path under ~/.m2/repository

If *you* would like it elsewhere, that's what `sonar.download.dir` is for.

Reply:
Those are the default cache directories for different OSes:

```
Mac OS X: ~/Library/Caches/<AppName>
Linux: ~/. cache/<AppName>
Win XP: C:\\Documents and Settings\\<username>\\Local Settings\\Application Data\\<AppAuthor>\\<AppName>\\Cache.
```

would it work if the we store the data under `<default cache dir>/cassandra/builds` ?

Reply:
@driftx I've applied your suggestion to use `tmp.dir`, but I've also added loading `~/.ant/build.properties` so that one can override globally `tmp.dir` for all projects.

Reply:
That works for me.